ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py

```diff
@@ -393,19 +375,6 @@ def _compute_system_metrics(
         logger.debug(f"Computing system metrics for {runner.table.__tablename__}")
         return self.system_metrics_computer.get_system_metrics(table=runner.table)
 
-    def _create_thread_safe_sampler(
-        self,
-        session,
-        table,
-    ):
-        """Create thread safe runner"""
-        if not hasattr(thread_local, "sampler"):
-            thread_local.sampler = self._get_sampler(
-                table=table,
-                session=session,
-            )
-        return thread_local.sampler
-
```

Comment:
we would need to test that tbh. I could use a hand validating if the error still exists. Since the sampler is an input arg now, does it change some internal state during the computation?

I guess before the issue was that we created a sampler object in each thread that needed to instantiate the same sample data. Right now it would not be the case right